api-reference/_hidden/GridBaseColumn/headerFilter/dataSource.md

```diff
@@ -14,242 +14,207 @@ Data source options.
 The widget's instance.
 
 ##### field(options.dataSource): DataSource_Options
-The header filter's data source.
+A [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) instance.
 
 ---
-This option accepts one of the following values:
+The **{WidgetName}** generates a header filter data source automatically based on column values. Use the **dataSource** option to change the generated data source or specify a custom data source.
 
-- **Array of Objects**      
-A simple JavaScript array containing a collection of plain objects. 
+### Specify a Custom Data Source
 
-    ---
-    #####jQuery
+To define a data source from scratch, set the **dataSource** option to an array of objects. Each object configures one header filter item and should have the following fields:
```

Comment:
"To define a data source from scratch" Can't you just replace it with 'to define a data source'?